### PR TITLE
Step 5

### PR DESCRIPTION
**Pull request by CaptainJNS**

### CODE REVIEW

File: app/admin/dashboard.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  content title: proc { I18n.t("active_admin.dashboard") } do
+    div class: "blank_slate_container", id: "dashboard_default_message" do
+      span class: "blank_slate" do
+        span I18n.t("active_admin.dashboard_welcome.welcome")
+        small I18n.t("active_admin.dashboard_welcome.call_to_action")
+      end
+    end
+
+    # Here is an example of a simple dashboard with columns and panels.
```
Comment (vatnajokull) [inline app/admin/dashboard.rb]: ???

File: app/models/book.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class Book < ApplicationRecord
   has_many :authors_books, dependent: :destroy
   has_many :authors, through: :authors_books
+  # accepts_nested_attributes_for :authors
```
Comment (vatnajokull) [inline app/models/book.rb]: ???

File: spec/features/admin/admin_authors_spec.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  it 'Admin can create an Author' do
+    click_link('New Author')
+
+    expect(page).to have_current_path(new_admin_author_path)
+
+    fill_in 'author[first_name]', with: author_attributes[:first_name]
+    fill_in 'author[last_name]', with: author_attributes[:last_name]
+
+    expect { click_button('Create Author'); sleep(1) }.to change { Author.count }.by(1)
```
Comment (vatnajokull) [inline spec/features/admin/admin_authors_spec.rb]: Do you really need `sleep` there?
Can you rewrite test and check that after creation you will be redirected to show page where title will be the name of Author.last?

File: spec/features/admin/admin_authors_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    expect(page).to have_current_path(admin_author_path(Author.last))
+  end
+
+  it 'Admin can delete the Author' do
+    expect { page.accept_confirm { click_link('Delete', match: :first) } }.to change(Author, :count).by(-1)
+  end
+
+  it 'Admin can edit the Author' do
+    new_f_name = FFaker::Name.first_name
```
Comment (vatnajokull) [inline spec/features/admin/admin_authors_spec.rb]: Could you please use more clear variable name?
`new_first_name` or just `first_name`

File: spec/features/admin/admin_authors_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    new_l_name = FFaker::Name.last_name
+    click_link('Edit', match: :first)
+
+    expect(page).to have_current_path(edit_admin_author_path(Author.last))
+
+    fill_in 'author[first_name]', with: new_f_name
+    fill_in 'author[last_name]', with: new_l_name
+    click_button('Update Author')
+    sleep(1)
```
Comment (vatnajokull) [inline spec/features/admin/admin_authors_spec.rb]: Please avoid sleep in feature tests where you can't use js

File: spec/features/admin/admin_authors_spec.rb (hunk 3 above)
Comment (vatnajokull) [reply spec/features/admin/admin_authors_spec.rb]: Why do you mark this comment as resolved and do nothing?

Comment (CaptainJNS) [reply spec/features/admin/admin_authors_spec.rb]: Sorry, I've changed it in next PR

File: app/admin/books.rb [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      span link_to('X', delete_book_image_admin_book_path(image.id),
+                   method: :delete, data: { confirm: 'Delete image?' })
+    end
+
+    f.actions
+  end
+
+  member_action :delete_book_image, method: :delete do
+    @image = ActiveStorage::Attachment.find(params[:id])
+    @image.purge_later
```
Comment (vatnajokull) [inline app/admin/books.rb]: Can we replace these two lines with one? `ActiveStorage::Attachment.find_by(id: params[:id])&.purge_later`

File: app/assets/javascripts/billings.coffee [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (vatnajokull) [inline app/assets/javascripts/billings.coffee]: Delete empty files

File: app/assets/javascripts/shippings.coffee [hunk 8]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (vatnajokull) [inline app/assets/javascripts/shippings.coffee]: Delete empty files

File: app/assets/stylesheets/billings.scss [hunk 9]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Billings controller here.
```
Comment (vatnajokull) [inline app/assets/stylesheets/billings.scss]: Delete empty files

File: app/assets/stylesheets/shippings.scss [hunk 10]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Shippings controller here.
```
Comment (vatnajokull) [inline app/assets/stylesheets/shippings.scss]: Delete empty files

File: app/controllers/billings_controller.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+class BillingsController < ApplicationController
```
Comment (vatnajokull) [inline app/controllers/billings_controller.rb]: Do you really need this controller?

File: app/controllers/shippings_controller.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+class ShippingsController < ApplicationController
```
Comment (vatnajokull) [inline app/controllers/shippings_controller.rb]: Do you really need this controller?

File: app/helpers/addresses_helper.rb [hunk 13]
```diff
@@ -0,0 +1,9 @@
+module AddressesHelper
```
Comment (vatnajokull) [inline app/helpers/addresses_helper.rb]: Do you use this helper?

**Pull request merged**